src/CG/CG.hpp

```diff
@@ -100,6 +100,7 @@ int convertBarycentricToFlagPolygon( std::size_t, double const *, double toleran
 void computeGeneralizedBarycentric( array3D const &, std::vector<array3D> const &, std::vector<double> &);
 void computeGeneralizedBarycentric( array3D const &, std::size_t, array3D const *, std::vector<double> &);
 void computeGeneralizedBarycentric( array3D const &, std::size_t, array3D const *, double *);
+void computeGeneralizedBarycentricDerivative( array3D const &, std::size_t, array3D const *, double *, std::array<double, 3> *);
```

Comment:
I don't like duplicating the code for the evaluation of the barycentric coordinates. It would be better to either modify the existing computeGeneralizedBarycentric functions to evaluate both barycentric coordinates and their derivative, or add a new set of functions to evaluate the barycentric coordinates derivative.

Reply:
I deleted completely the function. In the new version I avoid the usage of barycentric coordinates because I changed the algorithm of creating the surface over polygonal segments. Now the resulted surface has smoother behavior.